libraries/BluetoothSerial/src/BluetoothSerial.cpp

```diff
@@ -445,4 +662,128 @@ esp_err_t BluetoothSerial::register_callback(esp_spp_cb_t * callback)
+    esp_bt_gap_set_scan_mode(ESP_BT_SCAN_MODE_CONNECTABLE_DISCOVERABLE);
+    if (esp_bt_gap_start_discovery(ESP_BT_INQ_MODE_GENERAL_INQUIRY, INQ_LEN, INQ_NUM_RSPS) == ESP_OK) {
+        return waitForConnect(SCAN_TIMEOUT);
+    }
+    return false;
+}
+
+bool BluetoothSerial::connect(uint8_t remoteAddress[])
+{
+    if (!isReady(true, READY_TIMEOUT)) return false;
+    if (!remoteAddress) {
+        log_e("No remote address is provided");
+        return false; 
+    }
+    disconnect();
+    _remote_name[0] = 0;
+    _isRemoteAddressSet = true;
+    memcpy(_peer_bd_addr, remoteAddress, ESP_BD_ADDR_LEN);
+    log_i("master : remoteAddress");
+    if (esp_spp_start_discovery(_peer_bd_addr) == ESP_OK) {
+        return waitForConnect(READY_TIMEOUT);
+    }
+    return false;
+}
+
+bool BluetoothSerial::connect()
+{
+    if (!isReady(true, READY_TIMEOUT)) return false;
+    if (_isRemoteAddressSet){
+        disconnect();
+        // use resolved or set address first
+        log_i("master : remoteAddress");
+        if (esp_spp_start_discovery(_peer_bd_addr) == ESP_OK) {
+            return waitForConnect(READY_TIMEOUT);
+        }
+        return false;
+    } else if (_remote_name[0]) {
+        disconnect();
+        log_i("master : remoteName");
+        // will resolve name to address first - it may take a while
+        esp_bt_gap_set_scan_mode(ESP_BT_SCAN_MODE_CONNECTABLE_DISCOVERABLE);
+        if (esp_bt_gap_start_discovery(ESP_BT_INQ_MODE_GENERAL_INQUIRY, INQ_LEN, INQ_NUM_RSPS) == ESP_OK) {
+            return waitForConnect(SCAN_TIMEOUT);
+        }
+        return false;
+    }
+    log_e("Neither Remote name nor address was provided");
+    return false;
+}
+
+bool BluetoothSerial::disconnect() {
+    if (_spp_client) {
+        flush();
+        log_i("disconnecting");
+        if (esp_spp_disconnect(_spp_client) == ESP_OK) {
+            TickType_t xTicksToWait = READY_TIMEOUT / portTICK_PERIOD_MS;
+            return (xEventGroupWaitBits(_spp_event_group, SPP_DISCONNECTED, pdFALSE, pdTRUE, xTicksToWait) & SPP_DISCONNECTED);
+        }
+    }
+    return false;
+}
+
+bool BluetoothSerial::unpairDevice(uint8_t remoteAddress[]) {
+    if (isReady(false, READY_TIMEOUT)) {
+        log_i("removing bonded device");
+        return (esp_bt_gap_remove_bond_device(remoteAddress) == ESP_OK);
+    }
+    return false;
+}
+
+bool BluetoothSerial::connected(int timeout) {
+    return waitForConnect(timeout);
+}
+
+bool BluetoothSerial::isReady(bool checkMaster, int timeout) {
+    if (checkMaster && !_isMaster) {
+        log_e("Master mode is not active. Call begin(localName, true) to enable Master mode");
+        return false;
+    }
+    if (!btStarted()) {
+        log_e("BT is not initialized. Call begin() first");
+        return false;
+    }
+    TickType_t xTicksToWait = timeout / portTICK_PERIOD_MS;
+    return (xEventGroupWaitBits(_spp_event_group, SPP_RUNNING, pdFALSE, pdTRUE, xTicksToWait) & SPP_RUNNING);
```

Comment:
this will return the bit value, which might not be 1. You can just check if that is not equal to zero

Reply:
done